meta-balena-common/classes/balena-linux-firmware.bbclass

```diff
@@ -58,3 +60,14 @@ python __anonymous() {
 do_iwlwifi_firmware_clean[vardeps] += "IWLWIFI_FW_MIN_API_VARDEPS"
 addtask iwlwifi_firmware_clean after do_install before do_package
 addtask iwlwifi_firmware_clean after do_install before do_populate_sysroot
+
+fakeroot do_firmware_compression () {
+    if [ "${FIRMWARE_COMPRESSION}" = "1" ]; then
+        bbnote "Compressing firmware files"
+        find "${D}${nonarch_base_libdir}/firmware" -type l -exec sh -c 'target=$(readlink "$0"); ln -sf "${target}.xz" "$0"; mv "$0" "$0".xz' {} \;
+        find "${D}${nonarch_base_libdir}/firmware" -path "*/amd-ucode" -prune -o -type f -print -exec xz -C crc32 {} \;
```

Comment:
I've seen patches on the mailing list for supporting Zstd compression for firmware, so at some point it might be worth splitting the compression commands into variables for supporting other codecs.

Reply:
@jakogut why would we want it configurable? I mean, we can decide to switch to zstd or other, but what's the advantage of a variable?

Reply:
Different codecs have different characteristics as far as compression ratio, decompression speed, CPU and memory usage. I could see picking a different compression codec on a Pi Zero compared to a NUC, for example.

It's a moot point currently, however. The kernel only supports no compression or XZ at the moment.